modules/generator/storage/instance.go

```diff
@@ -49,7 +50,8 @@ func New(cfg *Config, tenant string, reg prometheus.Registerer, logger log.Logge
 	// subdirectory remote storage logs a scary error.
 	err := os.MkdirAll(filepath.Join(walDir, "wal"), 0o755)
 	if err != nil {
-		level.Warn(logger).Log("msg", "could not create directory for metrics WAL", "err", err)
+		level.Error(logger).Log("msg", "could not create directory for metrics WAL", "err", err)
```

Comment:
i'm fine with this if you want it, but does it make sense to log and return the error? or should we just rely on the caller to log the error?

Reply:
Yeah that's a good point. This error will get returned by the GRPC call so this error should show in the logs of the distributor and the metrics-generator. It will be a bit harder to parse as it gets wrapped in some more stuff but this shouldn't be an issue if you read it properly.